pkg/agent/agent.go

```diff
@@ -496,27 +487,77 @@ func (c *agentController) CreateAddOnPlacementScore() {
 			Name:      util.HostedClusterScoresResourceName,
 			Namespace: c.clusterName,
 		},
-		Status: clusterv1alpha1.AddOnPlacementScoreStatus{
-			Scores: items,
-		},
 	}
 
-	_, err = controllerutil.CreateOrUpdate(context.TODO(), c.hubClient, addOnPlacementScore, func() error { return nil })
+	_, err := controllerutil.CreateOrUpdate(context.TODO(), c.hubClient, addOnPlacementScore, func() error { return nil })
 	if err != nil {
 		// just log the error. it should not stop the rest of reconcile
 		c.log.Error(err, fmt.Sprintf("failed to create or update the addOnPlacementScore resource in %s", c.clusterName))
-		return
+		return err
```

Comment:
The just log comment might not match with the new behavior.

Reply:
I believe the just log is still the same as before, it just also returns the error externally to be used in case it fails when initializing the placement score on line 185. The reconcile doesn't stop.